rfd/0137-unified-resource-view.md

```diff
@@ -0,0 +1,151 @@
+---
+authors: Michael Myers (michael.myers@goteleport.com)
+state: draft
+---
+
+# RFD 137 - Unified Resource view for the web UI
+
+## What
+
+A new view on the web UI (and eventually Connect) that will display all available resources such as Nodes, Databases, Apps, etc, in a single unified way.
+
+![Screenshot 2023-07-25 at 4 18 08 PM](https://github.com/gravitational/teleport/assets/5201977/6953289d-ec3e-4eae-bdb9-10f3cfebfd7f)
+
+## Why
+
+Rather than separating all resources into their own individual pages, a Unified Resource page would help consolidate every kind of resource into a single, searchable view. This will help with discoverability and make searching across resources a lot better. This will also lay the foundation for a much more helpful "Dashboard" type page where we can have things such as Pinned Resources (coming right after this).
+
+## Details
+
+In order to get a searchable and filterable list of all resource kinds, we will create a new, in-memory store that will initialize on auth service start and only live as long as the service is up. The watcher will update the store any time it receives an event for any of the watched kinds. We will use a very similar setup for this as our in-memory backend, but instead of storing the resource in its textual representation required by the backend, we can store resources as their concrete type(e.g. types.ServerV2).
+
+### New Types
+`KindUnifiedResource` will be created and used as the `RequestType` for Unified Resource requests coming from the web. This will inform a few specific methods like `auth_with_roles.MakePaginatedResource` to check the response items type individually rather than collectively. 
```

Comment:
Is it still valid in the world where we have a separate API request? I think we left the `KindUnifiedResource`, but it became an internal implementation detail (one which, arguably, should be cleaned out so that it doesn't confuse future maintainers).